ImportDependencyDialog UX polish: compact Git panel, aligned rows, wider width

Jackherakovich4 opened the pull request:
• Removed multi-line “Preview” area (no longer useful) and deleted all
  obsolete helper methods/fields, resulting in a much shorter dialog.
• Re-built the Git section with a single GridBagLayout so
  “Repo URL:” and “Branch/Tag:” labels align perfectly.
• Renamed the validation button to “Load Tags & Branches”. • Clamped the branch/tag JComboBox to its preferred height and prevented
  the content panel from stealing extra vertical space (weighty = 0).
• Dropped the hard-coded minimum dialog size; instead pack() recomputes the
  natural size when the source type changes.
• Added 80 px of horizontal padding when Git mode is selected to give
  users more typing space for long repo URLs.
• Removed all leftover setPreferredSize() magic numbers and unused
  imports/fields to satisfy the style guide.
  
<img width="509" height="237" alt="Screenshot 2025-07-21 at 5 36 03 PM" src="https://github.com/user-attachments/assets/84e62807-c178-4db5-a3c9-53ae0acf5834" />

This ui doesn't feel perfect since it is a different size than the other two buttons, but it is a step up from the last one. Let me know if you have a better ui idea, I can't come up with anything.

bdragan commented on app/src/main/java/io/github/jbellis/brokk/gui/dialogs/ImportDependencyDialog.java:
```diff
@@ -591,4 +581,16 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
         });
     }
 
+    private static void deleteRecursively(Path path) throws IOException {
+        if (!Files.exists(path)) return;
+        try (Stream<Path> walk = Files.walk(path)) {
+            walk.sorted(Comparator.reverseOrder()).forEach(p -> {
+                try {
+                    Files.delete(p);
+                } catch (IOException e) {
+                    logger.warn("Failed to delete path during recursive cleanup: {}", p, e);
+                }
+            });
+        }
+    }
```
We already have a few of these throughout the code, maybe extract it into some util file so we can reuse the single one in the future?

bdragan replied:
I tried it and seems to be working fine. A couple of improvements:
- Repo URL input field is too small, I would place Load Tags & Branches button below it.
- Default branch (like `main` or `master`) should be selected by default.

Jackherakovich4 commented:
New Commit Addressing Comments:

•  Git-import section
   – Moved “Load Tags & Branches” to its own row so the repo URL field can
     expand to full width.
   – Gave the URL JTextField 30 columns and identical GridBag constraints
     (gridwidth = 2, weightx = 1, fill = HORIZONTAL) to keep the URL field,
     button, and Branch/Tag combo-box the same medium-long width.
   – Anchored all three controls WEST so any extra space appears on the right.

•  Default-branch heuristics
   – After loading refs, pre-select symbolic HEAD when present, otherwise fall
     back to “main” or “master”, then first branch or tag.

•  Code cleanup
   – Removed the dialog’s private deleteRecursively helper; now calls the
     shared `FileUtil.deleteRecursively` to avoid duplication.

bdragan commented:
> New Commit Addressing Comments:
> 
> • Git-import section – Moved “Load Tags & Branches” to its own row so the repo URL field can expand to full width. – Gave the URL JTextField 30 columns and identical GridBag constraints (gridwidth = 2, weightx = 1, fill = HORIZONTAL) to keep the URL field, button, and Branch/Tag combo-box the same medium-long width. – Anchored all three controls WEST so any extra space appears on the right.
> 
> • Default-branch heuristics – After loading refs, pre-select symbolic HEAD when present, otherwise fall back to “main” or “master”, then first branch or tag.
> 
> • Code cleanup – Removed the dialog’s private deleteRecursively helper; now calls the shared `FileUtil.deleteRecursively` to avoid duplication.

Only the new util class is in your latest commit, maybe some changes left unpushed in your local env?

Jackherakovich4 commented:
<img width="544" height="260" alt="Screenshot 2025-07-22 at 5 16 52 PM" src="https://github.com/user-attachments/assets/76bae516-c3b0-4786-bb8e-b11f53a5ab2b" />

This is the new appearance of the menu, let me know if this is not what you had in mind.
